### PR TITLE
games-*/*: EAPI7 bumps + eclass updates

### PR DESCRIPTION
**Pull request by mm1ke**
Hi,

Just some minor cleanups/EAPI bumps/eclass updates.
@ionenwks

### CODE REVIEW

File: games-action/moon-buggy/moon-buggy-1.0.51-r2.ebuild [hunk 1]
```diff
@@ -35,7 +35,7 @@ src_configure() {
 
 src_install() {
 	default
-	touch "${D}/var/games/${PN}/mbscore"
+	touch "${D}/var/games/${PN}/mbscore" || die
```
Comment (SoapGentoo) [inline games-action/moon-buggy/moon-buggy-1.0.51-r2.ebuild:38]:
```suggestion
	touch "${ED}/var/games/${PN}/mbscore" || die
```

File: games-board/cockatrice/cockatrice-20210126.ebuild [hunk 2]
```diff
@@ -3,14 +3,14 @@
 
 EAPI=7
 
-inherit cmake
+inherit cmake xdg-utils
```
Comment (SoapGentoo) [inline games-board/cockatrice/cockatrice-20210126.ebuild:6]:
```suggestion
inherit xdg cmake
```
and then kill the `pkg_*` phases

File: games-board/cockatrice/cockatrice-20210126.ebuild [hunk 3]
```diff
@@ -3,14 +3,14 @@
...
-DESCRIPTION="An open-source multiplatform software for playing card games over a network"
+DESCRIPTION="Open-source multiplatform software for playing card games over a network"
 HOMEPAGE="https://github.com/Cockatrice/Cockatrice"
 SRC_URI="https://github.com/Cockatrice/Cockatrice/archive/${MY_PV}.tar.gz -> ${P}.tar.gz"
 
-LICENSE="GPL-2"
+LICENSE="GPL-2+"
 SLOT="0"
 KEYWORDS="~amd64 ~x86"
 IUSE="+client +oracle test server"
```
Comment (SoapGentoo) [inline games-board/cockatrice/cockatrice-20210126.ebuild:16]: 1. move `test? ( dev-cpp/gtest )` into `DEPEND`
2. put the `S` after `SRC_URI`

File: games-puzzle/wmpuzzle/wmpuzzle-0.5.2-r1.ebuild [hunk 4]
```diff
@@ -1,25 +1,25 @@
...
-IUSE=""
 
 RDEPEND="x11-libs/libX11
 	x11-libs/libXext
 	x11-libs/libXpm"
 DEPEND="${RDEPEND}"
 
 PATCHES=( "${FILESDIR}"/${P}-gcc-10.patch )
-S=${WORKDIR}/${P}/src
+S="${WORKDIR}/${P}/src"
```
Comment (SoapGentoo) [inline games-puzzle/wmpuzzle/wmpuzzle-0.5.2-r1.ebuild:22]: after `SRC_URI`

**Review by SoapGentoo — CHANGES_REQUESTED** [review]
@mm1ke you used to be on IRC, maybe join us on Libera, I'd like to have a chat with you ;)

File: games-action/moon-buggy/moon-buggy-1.0.51-r2.ebuild [hunk 5]
```diff
@@ -1,29 +1,29 @@
...
+DEPEND="${RDEPEND}"
+BDEPEND="virtual/pkgconfig"
 
 src_prepare() {
 	default
 	sed -i \
 		-e '/$(DESTDIR)$(bindir)\/moon-buggy -c/d' \
 		Makefile.am || die
-	rm -f missing
+	rm -f missing || die
```
Comment (ionenwks) [inline games-action/moon-buggy/moon-buggy-1.0.51-r2.ebuild:26]: drop -f too, in src_prepare it's entirely predictable and -f makes no sense

File: games-action/moon-buggy/moon-buggy-1.0.51-r2.ebuild (hunk 1 above)
Comment (ionenwks) [reply games-action/moon-buggy/moon-buggy-1.0.51-r2.ebuild:38]: On that note, add ${EPREFIX} to /var/games in src_configure too, or else this wouldn't work right.

File: games-rpg/bass/bass-1.3-r1.ebuild [hunk 6]
```diff
@@ -29,7 +28,7 @@ src_install() {
 
 	insinto /usr/share/${PN}
 	doins sky.*
-	mv bass-cd-"${CD_VERSION}"/readme.txt readme-cd.txt
+	mv bass-cd-"${CD_VERSION}"/readme.txt readme-cd.txt || die
 	doins -r bass-cd-"${CD_VERSION}"/
 
 	make_wrapper bass "scummvm -f -p \"/usr/share/${PN}\" -q\$(scummvmGetLang.sh) sky" .
```
Comment (ionenwks) [inline games-rpg/bass/bass-1.3-r1.ebuild:34]: EPREFIX to both

**Review by ionenwks — COMMENTED** [review]
Just a few more things.

Also (for these) don't hesitate to do more misc re-styling when anything bothers you, most old games ebuilds are a mess.

**Comment by mm1ke** [comment]
@ionenwks @SoapGentoo thanks for the review. I've updated the ebuilds accordingly and i hope i didn't forget anything. :)
Also i'll keep on eye on the style and try to improve it.

File: games-board/cockatrice/cockatrice-20210126.ebuild [hunk 7]
```diff
@@ -3,14 +3,15 @@
 
 EAPI=7
 
-inherit cmake
+inherit cmake xdg
```
Comment (ionenwks) [inline games-board/cockatrice/cockatrice-20210126.ebuild:6]: Thanks, merged, but a note about this.

It's good to have sorted inherits but (as mentioned before) have to be careful when multiple inherits export the same function, Soap had them reversed in the suggestion for a reason.

Unlike xdg-utils, xdg exports src_prepare, and so does cmake. xdg's src_prepare isn't useful here (it's often skipped) but without cmake's this won't build, and then the last one inherited will be executed giving us xdg's only.
```
 * ERROR: games-board/cockatrice-20210126::gentoo failed (configure phase):
 *   FATAL: cmake_src_prepare has not been run
```
So, I amended it to `inherit xdg cmake`

In the event the ebuild defined its own src_prepare, this wouldn't have mattered. There's also other rare special cases like some java eclasses needing a specific order but generally not a worry.

Comment (mm1ke) [reply games-board/cockatrice/cockatrice-20210126.ebuild:6]: Ahh, thx for the fix and for the clarification.
Need to be careful with these next time.

**Pull request closed without merging**